test/integration/associations/belongs-to-many.test.js

```diff
@@ -1852,6 +1852,35 @@ describe(Support.getTestDialectTeaser('BelongsToMany'), function() {
       expect(Group.associations.MyUsers.through.model.rawAttributes.UserId).to.exist;
       expect(Group.associations.MyUsers.through.model.rawAttributes.GroupId).to.exist;
     });
+
+    it('handles constraints properly when through is a string', function() {
+      var User = this.sequelize.define('User', {})
+       , Group = this.sequelize.define('Group', {});
+
+      User.belongsToMany(Group, { as: 'MyGroups', through: 'group_user', onUpdate: 'RESTRICT', onDelete: 'SET NULL' });
+      Group.belongsToMany(User, { as: 'MyUsers', through: 'group_user', onUpdate: 'SET NULL', onDelete: 'RESTRICT' });
+
+      expect(Group.associations.MyUsers.through.model === User.associations.MyGroups.through.model);
+      expect(Group.associations.MyUsers.through.model.rawAttributes.UserId.onUpdate).to.equal('RESTRICT');
+      expect(Group.associations.MyUsers.through.model.rawAttributes.UserId.onDelete).to.equal('SET NULL');
+      expect(Group.associations.MyUsers.through.model.rawAttributes.GroupId.onUpdate).to.equal('SET NULL');
+      expect(Group.associations.MyUsers.through.model.rawAttributes.GroupId.onDelete).to.equal('RESTRICT');
+    });
+
+    it('handles constraints properly when through is a model', function() {
+      var User = this.sequelize.define('User', {})
+       , Group = this.sequelize.define('Group', {})
+       , UserGroup = this.sequelize.define('GroupUser', {}, {tableName: 'user_groups'});
+
+      User.belongsToMany(Group, { as: 'MyGroups', through: UserGroup, onUpdate: 'RESTRICT', onDelete: 'SET NULL' });
+      Group.belongsToMany(User, { as: 'MyUsers', through: UserGroup, onUpdate: 'SET NULL', onDelete: 'RESTRICT' });
+
+      expect(Group.associations.MyUsers.through.model === User.associations.MyGroups.through.model);
+      expect(Group.associations.MyUsers.through.model.rawAttributes.UserId.onUpdate).to.equal('RESTRICT');
+      expect(Group.associations.MyUsers.through.model.rawAttributes.UserId.onDelete).to.equal('SET NULL');
+      expect(Group.associations.MyUsers.through.model.rawAttributes.GroupId.onUpdate).to.equal('SET NULL');
+      expect(Group.associations.MyUsers.through.model.rawAttributes.GroupId.onDelete).to.equal('RESTRICT');
+    });
```

Comment:
Ok I'll move them. Shouldn't the ones above also be unit tests then?

I didn't as `npm install` failed for me on windows so I was relying on travis. Do you know if anyone has installed successfully on windows?

After the errors with `npm install` `npm test` results in `$COVERAGE was unexpected at this time.`